apt/method.go

```diff
@@ -66,31 +67,38 @@ type Method struct {
 
 type aptMethodConfig struct {
 	serviceAccountJSON, serviceAccountEmail string
+	debug                                   bool
 }
 
 // Run runs the method.
-func (m *Method) Run(ctx context.Context) {
+func (m *Method) Run(ctx context.Context) error {
 	m.writer.SendCapabilities()
 	for {
 		select {
 		case <-ctx.Done():
-			return
+			return nil
 		default:
 		}
 		msg, err := m.reader.ReadMessage(ctx)
-		if err != nil {
+		if err == ErrEmptyMessage {
 			continue
+		} else if err == io.EOF {
+			return nil
+		} else if err != nil {
+			return err
 		}
 		switch msg.code {
 		case 600:
 			m.handleAcquire(ctx, msg)
 		case 601:
 			m.handleConfigure(msg)
 		default:
-			// TODO: now write a test for this.
-			m.writer.Fail(fmt.Sprintf("Unsupported message code %d received from apt", msg.code))
+			if m.config.debug {
+				fmt.Fprintf(os.Stderr, "Unsupported message code %d received from apt", msg.code)
```

Comment:
Please write through m.writer, rather than using `fmt.Fprintf` to hardcode stderr.

Reply:
Though that may be true, making the plugin inconsistently hard-write to `stderr` sometimes and write to the user-configurable `writer` other times is bad practice. It'll work fine when the `writer` is also pointed at `stderr`, but when it isn't, errors will get logged confusingly to multiple places! ;)

If `stderr` is where logging should go, then `m.writer`'s default target should be set to be `stderr`.

A broad change of logging target is well outside your intended scope for this PR, so I'd encourage you to address it in a separate PR if you think it should be changed. :) Let's focus on fixing the bug at hand here, and _then_ revamp the logging separately.

Reply:
Although I wouldn't personally mind if the transport sent debug over `m.writer` as 101 (log) or 201 (status) packets, the built-in transports don't do this.

I do see what you mean though. Despite `m.writer` needing to be stdout for this program to function with apt, someone could potentially instantiate us with something else, such as is done in unit tests.

Given the above, I propose adding a `debugWriter` (`io.Writer`) on `Method` that `cmd/ar+https` sets to stderr. If that sounds OK, I'll split off the debug stuff into a separate PR.

Reply:
That sounds fine to me; sorry to have missed this message when it came in a couple of weeks ago. Just so long as all the code does the same thing universally, I'm happy.